span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGenerator.java

```diff
@@ -417,11 +426,11 @@ private static String removeTrailingSlash(String s) {
     return s.endsWith(SLASH) && s.length() > 1 ? s.substring(0, s.length() - 1) : s;
   }
 
-  private static Optional<String> getPathFromUriObject(String urlPath) {
+  private static Optional<String> getPathFromUrlObject(String urlPath) {
     try {
-      URI uri = new URI(urlPath);
-      return Optional.of(uri.getPath());
-    } catch (URISyntaxException e) {
+      URL url = new URL(dummyUrl, urlPath);
+      return Optional.of(url.getPath());
+    } catch (MalformedURLException e) {
```

Comment:
I know this is not part of the PR but I wonder if we should log this exception, at least at debug level. I can imagine an error in instrumentation where we pass invalid URL all the time and we don't have enough feedback on what happens. Instead if "http:google.com` shows up in the logs I'd know what is it.

Reply:
Sure. Added the logging statement at warning (instead of debug) level so as to track it effectively.